Tests/Benchmarks/BSONBenchmarks.swift

```diff
@@ -0,0 +1,113 @@
+let fullSize = 57.34
+
+let taskIterations = 10000
+
+final class EncodingBenchmarks: Benchmark {
+    static var allTests: [(String, (EncodingBenchmarks) -> () throws -> Void)] {
+        return [
+            ("testFlatEncoding", testFlatEncoding),
+            ("testDeepEncoding", testDeepEncoding),
+            ("testFullEncoding", testFullEncoding)
+        ]
+    }
+
+    // Read in the file, and then serialize its JSON by creating 
+    // a `Document`, which wraps the encoded data in a `bson_t`. 
+    // Repeat serialization 10,000 times. Unclear how helpful a benchmark
+    // this is as there is no C driver analog, so basically this is testing
+    // JSON parsing speed. 
+    func doEncodingTest(file: URL, size: Double, measureOpIterations: Int = 100) throws {
+        let jsonString = try String(contentsOf: file, encoding: .utf8)
+        let result = try measureOp({
+            for _ in 1...taskIterations {
+                _ = try Document(fromJSON: jsonString)
+            }
+        }, iterations: measureOpIterations)
+
+        printResults(time: result, size: size)
+    }
+
+    func testFlatEncoding() throws {
+        try doEncodingTest(file: flatBSONFile, size: flatSize)
+    }
+
+    func testDeepEncoding() throws {
+        try doEncodingTest(file: deepBSONFile, size: deepSize, measureOpIterations: 150)
+    }
+
+    func testFullEncoding() throws {
+        try doEncodingTest(file: fullBSONFile, size: fullSize)
+    }
+}
+
+final class DecodingBenchmarks: Benchmark {
+
+    static var allTests: [(String, (DecodingBenchmarks) -> () throws -> Void)] {
+        return [
+            ("testFlatDecoding", testFlatDecoding),
+            ("testDeepDecoding", testDeepDecoding),
+            ("testFullDecoding", testFullDecoding)
+        ]
+    }
+
+    // Recursively visit values 
+    func visit(_ value: BSONValue?) {
+        switch value {
+        // if a document or array, iterate to visit each value
+        case let val as Document:
+            for (_, v) in val {
+                self.visit(v)
+            }
+        case let val as [BSONValue]:
+            for v in val {
+                self.visit(v)
+            }
+        // otherwise, we are done
+        default:
+            break
+        }
+    }
+
+    // Read in the file, and then serialize its JSON by creating
+    // a `Document`, which wraps the encoded data in a `bson_t`. 
+    // "Deserialize" the data by recursively visiting each element.
+    // Repeat deserialization 10,000 times.
+    func doDecodingTest(file: URL, size: Double, measureOpIterations: Int = 100) throws {
+        let jsonString = try String(contentsOf: file, encoding: .utf8)
+        let document = try Document(fromJSON: jsonString)
+        let result = try measureOp({
+            for _ in 1...taskIterations {
+                self.visit(document)
+            }
+        }, iterations: measureOpIterations)
+
+        printResults(time: result, size: size)
+    }
+
+
```

Comment:
Nit and also kind of out of the scope of this PR but is tiny enough where I imagine it is worth doing now:
There is an extra newline here.

Reply:
fixed